### PR TITLE
ref(tracing): Pre-`traces_sampler` documentation additions

### PR DESCRIPTION
**Pull request by lobsterkatie**
A bunch of comments and docstrings which got added while I was working on https://github.com/getsentry/sentry-python/pull/863. Also adds data to both the `Span` and `Transaction` reprs, and changes a handful of variable names to be more accurate. No behavior changes.

### CODE REVIEW

File: sentry_sdk/tracing.py [hunk 1]
```diff
@@ -226,6 +229,14 @@ def continue_from_environ(
...
     ):
+        """
+        Create a Transaction with the given params, then add in data pulled from
+        the 'sentry-trace' header in the environ (if any) before returning the
+        Transaction.
+
+        If the 'sentry-trace' header is malformed or missing, just create and
+        return a Transaction instance with the given params.
+        """
         # type: (...) -> Transaction
```
Comment (flub) [inline sentry_sdk/tracing.py:240]: The linter doesn't seem to like this order of type comment after the docstring?

File: tests/integrations/stdlib/test_httplib.py [hunk 2]
```diff
@@ -4,13 +4,17 @@
 import pytest
 
 try:
+    # py3
```
Comment (flub) [inline tests/integrations/stdlib/test_httplib.py:7]: I've seem people argue for `if PY3K:` kind of variations for these to make this clearer.  Anyway, this is fine by me - i don't know if we have internal coding standards which have other opinions.

**Review by flub — APPROVED** [review]

File: sentry_sdk/tracing.py (hunk 1 above)
Comment (lobsterkatie) [reply sentry_sdk/tracing.py:245]: Oh, _that's_ what it was mad about! The problematic ordering predates me, so I figured it was just the linter not being 100% on board with type hints. Fixed now, thanks.

File: tests/integrations/stdlib/test_httplib.py (hunk 2 above)
Comment (lobsterkatie) [reply tests/integrations/stdlib/test_httplib.py:7]: Yeah, that might be better, but for now my goal was only to mark which was which.

File: sentry_sdk/tracing.py (hunk 1 above)
Comment (lobsterkatie) [reply sentry_sdk/tracing.py]: Ugh. After a lot of poking, I figured out what the problem is: `mypy` wants the type annotations before the docstring, whereas `pylance` (the VSCode Python engine) very much wants them after. So now my editor is all kinds of mad, but at least CI is passing.

**Pull request merged**